samples/CMakeLists.txt

```diff
@@ -80,23 +80,34 @@ else ()
     RENAME README.md)
 endif ()
 
+# Determine if SGX DCAP libraries are available in the test system.
```

Comment:
This is still an awkward way to run the tests, because it has to assume that the system that builds the OE SDK project and ctests is also the same system that the test run on, which the removal of the HAS_QUOTE_PROVIDER flag elsewhere would otherwise help decouple. (i.e. it enables forward looking scenarios like allowing build on normal GP machine pools, then transfer to SGX & DCAP enabled machine to test).

Instead of making this a build variable, have we considered making this an environment variable checked directly in samples/test-samples.cmake? It's something that we already do with the `OE_SIMULATION` flag, so it seems appropriate that we should just move this one out to a run time setting we can tweak in CI/CD directly.

Reply:
In principle, re-running cmake with the right flag for the target platform before triggering ctest ought to be enough to address those cases, since it acts incrementally, assuming no rebuild is needed (but then transferring the binaries wouldn't work). The nice thing about cmake flags compared to env vars is that they're more discoverable.

Reply:
I'd originally tried putting the logic for inferring HAS_QUOTE_PROVIDER in test-samples.cmake. That would've allowed the samples test to automatically select the appropriate list of samples to be tested. 
I gave up on that since `find_package` does not work in cmake scripts (the -P option).
I will try to find a work around.

Though an environment variable allows configuration at test execution time; it still has to be set up correctly which has to be done either by the user (which implies more responsibility on the user for testing attestaton samples) or by the cmake system (which implies a coupling between the build machine and the samples list).

Reply:
Yeah, I think it's useful to consider the scenarios here:
1. **SDK user.** This CMakeLists.txt doesn't affect them, they always have to figure out their configuration and whether to build/run individual samples or not.
2. **OE SDK committer.** This PR preserves convenience for them where we consider the building and testing on the same device configuration as the primary use case.
   - I think I'm also much more willing to assume a default config in these scenarios, as we do with `OE_SIMULATION`. e.g we could define the environment var as the inverse `NO_DCAP_PROVIDER` so it doesn't have to explicitly set if we expect most committers to build and test with the provider. It's also a set-once property, which isn't significantly onerous in my mind and actually easier to toggle for experimental purposes than installing/uninstalling a DCAP provider lib.
   - I'll also note that the `find_package` check only indicates that a DCAP lib is installed, not that it is either compatible with OE SDK or will run, it's always the user's responsibility to understand that anyway. It's a reasonable assumption though that if you installed it, you've set up the system that it will run. 
      - A different approach may be for CMake to infer this and set the environmental variable on your behalf, if it hasn't been previously set, and to warn if it is set to a value counter to what cmake detects.  
3. **CI/CD**. This is more likely to use the build and test on different environments, and for that use case, I disagree with @achamayou's position that rerunning cmake is fine; that would require cloning both source and output trees to the test device which is largely unnecessary otherwise.

Reply:
@CodeMonkeyLeet 
Can you look at the latest implementation?
- It uses cmake itself to determine whether the target system has DCAP libraries.
  As was the case with HAS_QUOTE_PROVIDER=on/off, no checks are made as to whether the DCAP libraries is compatible with the SDK or not.
- The build can be transferred to another system and the DCAP determination will be done on that system.

Reply:
>     3. **CI/CD**. This is more likely to use the build and test on different environments, and for that use case, I disagree with @achamayou's position that rerunning cmake is fine; that would require cloning both source and output trees to the test device which is largely unnecessary otherwise.

cmake creates dependency rules between build artifacts and the sources for which the artifact is generated. This allows it to trigger the rebuild of an artifact if the source has been deemed to have changed.

What this implies is that we'd have to copy both the source and output trees from the build system on to the test system.
Additionally, the time stamps as well as the full file paths have to be preserved in order to be able to ctest on a different machine.